runtime/wasm/src/host_exports.rs

```diff
@@ -660,7 +660,7 @@ impl<C: Blockchain> HostExports<C> {
 
         // Remember that we need to create this data source
         state.push_created_data_source(DataSourceTemplateInfo {
-            template,
+            template: template.clone(),
```

Comment:
Do you understand why moving the bound from the trait to the associated type resulted in this being necessary? This is confusing to me.

Reply:
This crate doesn't compile for me locally, it might not be necessary. Will look into it